packages/core/src/install/createPeerDependencyPatcher.ts

```diff
@@ -21,7 +21,10 @@ export default function (
         if (peerDependencyRules.allowedVersions[peerName] === '*') {
           pkg.peerDependencies![peerName] = '*'
         } else {
-          pkg.peerDependencies![peerName] += ` || ${peerDependencyRules.allowedVersions[peerName]}`
+          const allowedVersion = ` || ${peerDependencyRules.allowedVersions[peerName]}`
```

Comment:
what about `${peerDependencyRules.allowedVersions[peerName]} ||` ?

Reply:
I was aiming to remove duplication of the expanded version range but leave any base versions alone, which means that I was only targeting duplicates after the first version, which would have the ` || ` as a prefix instead of a postfix.

That could still leave one duplicate if the core package's peer dependency version was identical to the one you're expanding, but that would be deterministic and not cause conflicts like a variable level of duplication.

If you prefer I can easily replace both.

Reply:
in that case, you should probably use `.endsWith` instead of `.includes`

Reply:
Refactored it a bit to handle the edge cases, and added test cases for not extending the same single version, or if you set a range of allowed versions. It'll be a bit slower but safer.